projects/storefrontlib/src/cms-components/myaccount/update-email/update-email.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import {
+  AuthService,
+  GlobalMessageService,
+  GlobalMessageType,
+  RoutingService,
+  UserService,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { CustomFormValidators } from '../../../shared/utils';
+import { first } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UpdateEmailService implements OnDestroy {
+  protected subscriptions = new Subscription();
+  isUpdating$: Observable<
+    boolean
+  > = this.userService.getUpdateEmailResultLoading();
+
+  constructor(
+    protected routingService: RoutingService,
+    protected globalMessageService: GlobalMessageService,
+    protected userService: UserService,
+    protected authService: AuthService
+  ) {}
+
+  form: FormGroup = new FormGroup(
+    {
+      email: new FormControl('', [
+        Validators.required,
+        CustomFormValidators.emailValidator,
+      ]),
+      confirmEmail: new FormControl('', [Validators.required]),
+      password: new FormControl('', [Validators.required]),
+    },
+    {
+      validators: CustomFormValidators.emailsMustMatch('email', 'confirmEmail'),
+    }
+  );
+
+  reset(): void {
+    this.userService.resetUpdateEmailResultState();
+    this.form.reset();
+  }
+
+  ngOnDestroy(): void {
```

Comment:
I was hoping to omit the ngDestroy and infinite subscriptions in this service by:
- use the `first(Boolean)` operator, which stops subscribing after the first succesful emission is passed
- disable the form in save() method, and re-enable  in the success